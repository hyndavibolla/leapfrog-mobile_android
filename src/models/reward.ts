export enum SlideObjectType {
  BRAND = 'brands',
  CATEGORY = 'categories'
}

export interface IBrandDetail {
  brandId: string;
  brandName: string;
}

export interface ICategoryConfig {
  id: string;
  brands: IBrandDetail[];
}

export interface IRewardConfig {
  categories: ICategoryConfig[];
  limits: {
    minAmount: number;
    maxAmount: number;
    duration: string;
  };
}

export interface ICardValueConfig {
  increment: number;
  minValue: number;
  maxValue: number;
  variableLoadSupported: boolean;
  denominations?: number[];
}

export interface IRedemptionConfig {
  disclaimer: string;
  kind: string;
  methods: { info: string; kind: string }[];
}

export interface ICategory {
  name: string;
  availableBrands?: number;
}

export interface IBrand {
  id?: string; // added on the client
  brandName: string;
  iconUrl: string;
  legalTerms: string;
  description: string;
  cardValueConfig: ICardValueConfig;
  redemptionConfigs: IRedemptionConfig;
  sortedPosition: number;
  categories: ICategory[];
  backOfCardUrl?: string;
  faceplateUrl?: string;
}

export type ISlideObjectAttribute = IBrand | ICategory;

export interface ISlideObject {
  id: string; // might be an UUID or an enum of some kind
  type: SlideObjectType;
  attributes: ISlideObjectAttribute;
}

export const getFallbackBrand = (id: string): IBrand => ({
  id,
  brandName: '...',
  iconUrl: '',
  legalTerms: '...',
  description: '...',
  cardValueConfig: {
    increment: 1,
    minValue: 0,
    maxValue: 0,
    variableLoadSupported: false,
    denominations: []
  },
  redemptionConfigs: {
    disclaimer: '...',
    kind: '...',
    methods: []
  },
  sortedPosition: 0,
  categories: []
});
