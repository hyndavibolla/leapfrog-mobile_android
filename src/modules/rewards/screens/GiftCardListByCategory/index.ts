export { default as GiftCardListByCategory } from './GiftCardListByCategory';
