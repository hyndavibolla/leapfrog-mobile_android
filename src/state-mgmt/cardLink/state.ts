import { ICardLinkOffer, ILinkedCard, ILocation } from '_models/cardLink';

export interface ICardLinkState {
  userId: string;
  offers: ICardLinkOffer[];
  linkedCardsList: ILinkedCard[];
  lastSearchedLocation: ILocation;
  isLocalOfferFailed: boolean;
  hasCalledLocalOffer: boolean; // to check if at least one time has been called getLocalOffers
  routeToActivateLocalOffer: string; // to activate the local offer after activated a new credit card
}

export const initialState: ICardLinkState = {
  userId: null,
  offers: [],
  linkedCardsList: [],
  lastSearchedLocation: null,
  isLocalOfferFailed: false,
  hasCalledLocalOffer: false,
  routeToActivateLocalOffer: ''
};
