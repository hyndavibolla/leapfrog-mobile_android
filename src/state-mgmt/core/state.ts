import { Props as ModalProps } from '_components/Modal';
import { Props as ToastProps } from '_components/Toast';
import { IExpirePointsBannerDataSet, TooltipKey } from '_models/general';

export interface ICoreState {
  isDeepLinkListenerReady: boolean;
  isAppReady: boolean;
  isConnected: boolean;
  toastList: { key: string; props: ToastProps }[];
  modalList: { key: string; props: ModalProps }[];
  deepLink: { route: string; params: Object };
  lastRouteKey: string;
  lastRouteParams: object;
  routeHistory: string[];
  hasSeenOnboarding: boolean;
  viewedTooltipList: TooltipKey[];
  dismissedTooltipList: TooltipKey[];
  hasSeenMissionModal: boolean;
  expirePointsBannerDataSet: IExpirePointsBannerDataSet;
  roToken: string;
  marketingId: string;
  showForcedUpdateScreen: boolean;
  isTutorialVisible: boolean;
  isTutorialAvailable: boolean;
  tutorialFrom: string;
  experimentsConfig: object;
}

export const initialState: ICoreState = {
  isDeepLinkListenerReady: false,
  isAppReady: false,
  isConnected: true,
  toastList: [],
  modalList: [],
  deepLink: null,
  lastRouteKey: null,
  lastRouteParams: undefined,
  routeHistory: [],
  hasSeenOnboarding: null,
  viewedTooltipList: [],
  dismissedTooltipList: [],
  hasSeenMissionModal: false,
  expirePointsBannerDataSet: { lastAcceptedDate: 0, lastPointsDate: 0 },
  roToken: null,
  marketingId: undefined,
  showForcedUpdateScreen: false,
  isTutorialVisible: false,
  isTutorialAvailable: false,
  tutorialFrom: '',
  experimentsConfig: {}
};
