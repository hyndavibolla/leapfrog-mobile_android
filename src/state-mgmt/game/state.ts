import { GameModel } from '../../models';

export interface IGameState {
  current: GameModel.IGame;
  historical: { totalLifetimeEarnedPoints: number; pointByMonthMap: { [month: string]: number } };
}

export const initialState: IGameState = {
  current: {
    balance: {
      availablePoints: 0,
      pendingPoints: 0,
      memberOwnPointsToExpire: []
    },
    memberships: {
      userHasSywCard: false,
      userIsSywMaxMember: false,
      userIsAllowedToUseSywMax: true,
      sywCardLastFour: '1234'
    },
    missions: {
      pointsPerCent: 0
    },
    lastUpdatedAt: 0
  },
  historical: {
    totalLifetimeEarnedPoints: 0,
    pointByMonthMap: {}
  }
};
