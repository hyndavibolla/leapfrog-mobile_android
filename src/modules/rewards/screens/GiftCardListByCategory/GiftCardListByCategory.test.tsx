import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import GiftCardListByCategory, { Props } from './GiftCardListByCategory';

import { KnownSlideObjectSearchKey } from '_state_mgmt/reward/state';
import { getInitialState } from '_state_mgmt/GlobalState';
import { RewardModel } from '_models';
import { Deps, IGlobalState } from '_models/general';

import { getMockDeps } from '_test_utils/getMockDeps';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { getRewardConfig_1, getSlideBrand_1, getSlideBrand_2, getSlideBrand_3, getSlideCategoryIdList } from '_test_utils/entities';
import { listToMap } from '_utils/listToMap';
import { ROUTES } from '_constants/routes';

describe('GiftCardListByCategory', () => {
  let props: Props;
  let deps: Deps;
  let initState: IGlobalState;

  beforeEach(() => {
    initState = getInitialState();
    props = {
      navigation: {
        setParams: jest.fn(),
        isFocused: jest.fn(() => false),
        navigate: jest.fn(),
        setOptions: jest.fn().mockImplementation(({ header }) => header())
      } as any,
      route: { params: { searchKey: KnownSlideObjectSearchKey.FULL_BRAND_SEARCH_RESULTS } }
    };
    deps = getMockDeps();

    initState.reward.config = getRewardConfig_1();
    initState.reward.slideObjectSearchMap[KnownSlideObjectSearchKey.FULL_BRAND_SEARCH_RESULTS] = [
      getSlideBrand_1().id,
      getSlideBrand_2().id,
      getSlideBrand_3().id
    ];
    initState.reward.slideObjectMapByType[RewardModel.SlideObjectType.BRAND] = listToMap([getSlideBrand_1(), getSlideBrand_2(), getSlideBrand_3()]);
    initState.reward.slideCategoryIdList = getSlideCategoryIdList();
  });

  it('should render with incomplete data', async () => {
    initState.reward.slideObjectSearchMap[KnownSlideObjectSearchKey.FULL_BRAND_SEARCH_RESULTS] = null;

    const { toJSON } = renderWithGlobalContext(<GiftCardListByCategory {...props} />, deps, initState);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with data', async () => {
    props.route.params = null;
    initState.reward.slideObjectSearchMap[KnownSlideObjectSearchKey.FULL_BRAND_SEARCH_RESULTS] = [
      getSlideBrand_1().id,
      getSlideBrand_2().id,
      getSlideBrand_3().id,
      null
    ];

    const { toJSON } = renderWithGlobalContext(<GiftCardListByCategory {...props} />, deps, initState);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have a scroll to top button', async () => {
    const { queryByTestId, getByTestId } = renderWithGlobalContext(<GiftCardListByCategory {...props} />, deps, initState);
    expect(queryByTestId('gift-card-list-category-scroll-btn')).toBeNull();

    fireEvent.scroll(getByTestId('gift-card-list-category-scroll'), {
      nativeEvent: {
        contentOffset: { y: 500 },
        contentSize: { height: 500, width: 100 },
        layoutMeasurement: { height: 100, width: 100 }
      }
    });

    expect(getByTestId('gift-card-list-category-scroll-btn')).toBeTruthy();
    fireEvent.press(getByTestId('gift-card-list-category-scroll-btn'));
  });

  it('should filter', async () => {
    props.route.params.searchCategoryIdList = [getSlideCategoryIdList()[0]];

    const { findAllByTestId } = renderWithGlobalContext(<GiftCardListByCategory {...props} />, deps, initState);
    expect(await findAllByTestId('small-gift-card-container')).toHaveLength(3);
  });

  it('should navigate to details when a card is pressed', async () => {
    const { getAllByTestId } = renderWithGlobalContext(<GiftCardListByCategory {...props} />, deps, initState);

    fireEvent.press(getAllByTestId('small-gift-card-container')[0]);
    expect(props.navigation.navigate).toBeCalledWith(ROUTES.GIFT_CARD_DETAIL, { brandId: getSlideBrand_1().id });
  });

  it('should show the searched text', async () => {
    props.route.params.searchText = 'text';

    const { findByTestId } = renderWithGlobalContext(<GiftCardListByCategory {...props} />, deps, initState);
    expect(findByTestId('gift-card-list-category-searched-text')).toBeTruthy();
  });
});
