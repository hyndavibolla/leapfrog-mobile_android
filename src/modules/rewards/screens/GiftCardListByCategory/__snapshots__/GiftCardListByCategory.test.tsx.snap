// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftCardListByCategory should render with data 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingTop": 32,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    ListFooterComponent={
      <View
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    }
    ListFooterComponentStyle={
      Object {
        "marginBottom": 40,
      }
    }
    ListHeaderComponent={
      <Memo(Text)
        accessibilityLabel="gift-card-list-category-searched-text"
        accessible={true}
        style={
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 16,
            "marginBottom": 20,
          }
        }
        testID="gift-card-list-category-searched-text"
      >
        
         Gift Cards
      </Memo(Text)>
    }
    accessibilityLabel="gift-card-list-category-scroll"
    accessible={true}
    contentContainerStyle={false}
    data={
      Array [
        Object {
          "brandName": "Burgers For Days",
          "cardValueConfig": Object {
            "increment": 1,
            "maxValue": 50000,
            "minValue": 500,
            "variableLoadSupported": true,
          },
          "categories": Array [
            Object {
              "name": "Entertainment",
            },
            Object {
              "name": "Travel",
            },
            Object {
              "name": "Automotive",
            },
          ],
          "description": "Buy stuff, buy lots of it.",
          "iconUrl": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
          "id": "754338d5-7c5e-4785-bd05-67cd46545df9",
          "legalTerms": "This card can be used to purchase merchandise only in stores",
          "redemptionConfigs": Object {
            "disclaimer": "Can only use 2 cards",
            "kind": "IN_STORE_OR_ONLINE_ONLY",
            "methods": Array [
              Object {
                "info": "Enter the gift card info online",
                "kind": "ENTER_ONLINE",
              },
              Object {
                "info": "Show the barcode to the cashier",
                "kind": "SHOW_BARCODE",
              },
            ],
          },
          "sortedPosition": 2,
        },
        Object {
          "brandName": "Durum For Weeks",
          "cardValueConfig": Object {
            "increment": 1,
            "maxValue": 50000,
            "minValue": 500,
            "variableLoadSupported": true,
          },
          "categories": Array [
            Object {
              "name": "Entertainment",
            },
          ],
          "description": "Buy stuff, buy lots of it.",
          "iconUrl": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
          "id": "67cd46545df9-754338d5-7c5e-4785-bd05",
          "legalTerms": "This card can be used to purchase merchandise only in stores",
          "redemptionConfigs": Object {
            "disclaimer": "Can only use 2 cards",
            "kind": "IN_STORE_OR_ONLINE_ONLY",
            "methods": Array [
              Object {
                "info": "Enter the gift card info online",
                "kind": "ENTER_ONLINE",
              },
              Object {
                "info": "Show the barcode to the cashier",
                "kind": "SHOW_BARCODE",
              },
            ],
          },
          "sortedPosition": 2,
        },
        Object {
          "brandName": "Durum For Years",
          "cardValueConfig": Object {
            "increment": 1,
            "maxValue": 50000,
            "minValue": 500,
            "variableLoadSupported": true,
          },
          "categories": Array [
            Object {
              "name": "Entertainment",
            },
          ],
          "description": "Buy stuff, buy lots of it.",
          "iconUrl": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
          "id": "67cd46545df9-7c5e-4785-bd05-754338d5",
          "legalTerms": "This card can be used to purchase merchandise only in stores",
          "redemptionConfigs": Object {
            "disclaimer": "Can only use 2 cards",
            "kind": "IN_STORE_OR_ONLINE_ONLY",
            "methods": Array [
              Object {
                "info": "Enter the gift card info online",
                "kind": "ENTER_ONLINE",
              },
              Object {
                "info": "Show the barcode to the cashier",
                "kind": "SHOW_BARCODE",
              },
            ],
          },
          "sortedPosition": 2,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "alignSelf": "center",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
        "width": "100%",
      }
    }
    testID="gift-card-list-category-scroll"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessibilityLabel="gift-card-list-category-searched-text"
          accessible={true}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#262C2F",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 16,
                "marginBottom": 20,
              },
            ]
          }
          testID="gift-card-list-category-searched-text"
        >
          
           Gift Cards
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginBottom": 8,
            }
          }
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessibilityLabel="small-gift-card-container"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="small-gift-card-container"
            >
              <View
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "elevation": 3,
                        "shadowColor": "#262C2F",
                        "shadowOffset": Object {
                          "height": 4,
                          "width": 0,
                        },
                        "shadowOpacity": 0.1,
                      },
                      Object {
                        "borderBottomLeftRadius": 8,
                        "borderBottomRightRadius": 8,
                        "borderTopLeftRadius": 8,
                        "borderTopRightRadius": 8,
                        "flexDirection": "column",
                        "justifyContent": "space-between",
                        "paddingBottom": 20,
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "paddingTop": 20,
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#FFFFFF",
                          "flexDirection": "row",
                          "height": 90,
                          "justifyContent": "center",
                          "paddingBottom": 20,
                          "paddingLeft": 20,
                          "paddingRight": 20,
                          "paddingTop": 20,
                        },
                        false,
                        Object {
                          "width": "100%",
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 60,
                          "width": 60,
                        },
                        Object {
                          "backgroundColor": "#E2E3E3",
                          "borderBottomLeftRadius": 30,
                          "borderBottomRightRadius": 30,
                          "borderTopLeftRadius": 30,
                          "borderTopRightRadius": 30,
                          "height": 60,
                          "width": 60,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "alignItems": "center",
                            "borderBottomLeftRadius": 60,
                            "borderBottomRightRadius": 60,
                            "borderTopLeftRadius": 60,
                            "borderTopRightRadius": 60,
                            "height": "100%",
                            "justifyContent": "center",
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <FastImageView
                        accessibilityLabel="brand-logo-image"
                        accessible={true}
                        onFastImageError={[Function]}
                        onFastImageLoad={[Function]}
                        onFastImageLoadEnd={[Function]}
                        onFastImageLoadStart={[Function]}
                        resizeMode="cover"
                        source={
                          Object {
                            "priority": "high",
                            "uri": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                        testID="brand-logo-image"
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "justifyContent": "center",
                        },
                        Object {
                          "marginLeft": 20,
                        },
                        false,
                      ]
                    }
                  >
                    <Text
                      accessibilityLabel="small-gift-card-brand-name"
                      accessible={true}
                      ellipsizeMode="tail"
                      font="SFProDisplay-Bold"
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "color": "#909BA0",
                            "fontFamily": "SFProDisplay-Bold",
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                          Array [
                            Object {
                              "color": "#262C2F",
                              "fontSize": 18,
                            },
                            false,
                          ],
                        ]
                      }
                      testID="small-gift-card-brand-name"
                    >
                      Burgers For Days
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginBottom": 8,
            }
          }
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessibilityLabel="small-gift-card-container"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="small-gift-card-container"
            >
              <View
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "elevation": 3,
                        "shadowColor": "#262C2F",
                        "shadowOffset": Object {
                          "height": 4,
                          "width": 0,
                        },
                        "shadowOpacity": 0.1,
                      },
                      Object {
                        "borderBottomLeftRadius": 8,
                        "borderBottomRightRadius": 8,
                        "borderTopLeftRadius": 8,
                        "borderTopRightRadius": 8,
                        "flexDirection": "column",
                        "justifyContent": "space-between",
                        "paddingBottom": 20,
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "paddingTop": 20,
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#FFFFFF",
                          "flexDirection": "row",
                          "height": 90,
                          "justifyContent": "center",
                          "paddingBottom": 20,
                          "paddingLeft": 20,
                          "paddingRight": 20,
                          "paddingTop": 20,
                        },
                        false,
                        Object {
                          "width": "100%",
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 60,
                          "width": 60,
                        },
                        Object {
                          "backgroundColor": "#E2E3E3",
                          "borderBottomLeftRadius": 30,
                          "borderBottomRightRadius": 30,
                          "borderTopLeftRadius": 30,
                          "borderTopRightRadius": 30,
                          "height": 60,
                          "width": 60,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "alignItems": "center",
                            "borderBottomLeftRadius": 60,
                            "borderBottomRightRadius": 60,
                            "borderTopLeftRadius": 60,
                            "borderTopRightRadius": 60,
                            "height": "100%",
                            "justifyContent": "center",
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <FastImageView
                        accessibilityLabel="brand-logo-image"
                        accessible={true}
                        onFastImageError={[Function]}
                        onFastImageLoad={[Function]}
                        onFastImageLoadEnd={[Function]}
                        onFastImageLoadStart={[Function]}
                        resizeMode="cover"
                        source={
                          Object {
                            "priority": "high",
                            "uri": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                        testID="brand-logo-image"
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "justifyContent": "center",
                        },
                        Object {
                          "marginLeft": 20,
                        },
                        false,
                      ]
                    }
                  >
                    <Text
                      accessibilityLabel="small-gift-card-brand-name"
                      accessible={true}
                      ellipsizeMode="tail"
                      font="SFProDisplay-Bold"
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "color": "#909BA0",
                            "fontFamily": "SFProDisplay-Bold",
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                          Array [
                            Object {
                              "color": "#262C2F",
                              "fontSize": 18,
                            },
                            false,
                          ],
                        ]
                      }
                      testID="small-gift-card-brand-name"
                    >
                      Durum For Weeks
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginBottom": 8,
            }
          }
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessibilityLabel="small-gift-card-container"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="small-gift-card-container"
            >
              <View
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "elevation": 3,
                        "shadowColor": "#262C2F",
                        "shadowOffset": Object {
                          "height": 4,
                          "width": 0,
                        },
                        "shadowOpacity": 0.1,
                      },
                      Object {
                        "borderBottomLeftRadius": 8,
                        "borderBottomRightRadius": 8,
                        "borderTopLeftRadius": 8,
                        "borderTopRightRadius": 8,
                        "flexDirection": "column",
                        "justifyContent": "space-between",
                        "paddingBottom": 20,
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "paddingTop": 20,
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#FFFFFF",
                          "flexDirection": "row",
                          "height": 90,
                          "justifyContent": "center",
                          "paddingBottom": 20,
                          "paddingLeft": 20,
                          "paddingRight": 20,
                          "paddingTop": 20,
                        },
                        false,
                        Object {
                          "width": "100%",
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 60,
                          "width": 60,
                        },
                        Object {
                          "backgroundColor": "#E2E3E3",
                          "borderBottomLeftRadius": 30,
                          "borderBottomRightRadius": 30,
                          "borderTopLeftRadius": 30,
                          "borderTopRightRadius": 30,
                          "height": 60,
                          "width": 60,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "alignItems": "center",
                            "borderBottomLeftRadius": 60,
                            "borderBottomRightRadius": 60,
                            "borderTopLeftRadius": 60,
                            "borderTopRightRadius": 60,
                            "height": "100%",
                            "justifyContent": "center",
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <FastImageView
                        accessibilityLabel="brand-logo-image"
                        accessible={true}
                        onFastImageError={[Function]}
                        onFastImageLoad={[Function]}
                        onFastImageLoadEnd={[Function]}
                        onFastImageLoadStart={[Function]}
                        resizeMode="cover"
                        source={
                          Object {
                            "priority": "high",
                            "uri": "https://s3.amazonaws.com/raise-content/ibi/GameStop-Logo.png",
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                        testID="brand-logo-image"
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "justifyContent": "center",
                        },
                        Object {
                          "marginLeft": 20,
                        },
                        false,
                      ]
                    }
                  >
                    <Text
                      accessibilityLabel="small-gift-card-brand-name"
                      accessible={true}
                      ellipsizeMode="tail"
                      font="SFProDisplay-Bold"
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "color": "#909BA0",
                            "fontFamily": "SFProDisplay-Bold",
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                          Array [
                            Object {
                              "color": "#262C2F",
                              "fontSize": 18,
                            },
                            false,
                          ],
                        ]
                      }
                      testID="small-gift-card-brand-name"
                    >
                      Durum For Years
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "marginBottom": 40,
          }
        }
      >
        <View
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`GiftCardListByCategory should render with incomplete data 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingTop": 32,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    ListFooterComponent={
      <View
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    }
    ListFooterComponentStyle={
      Object {
        "marginBottom": 40,
      }
    }
    ListHeaderComponent={
      <Memo(Text)
        accessibilityLabel="gift-card-list-category-searched-text"
        accessible={true}
        style={
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 16,
            "marginBottom": 20,
          }
        }
        testID="gift-card-list-category-searched-text"
      >
        
         Gift Cards
      </Memo(Text)>
    }
    accessibilityLabel="gift-card-list-category-scroll"
    accessible={true}
    contentContainerStyle={
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "alignSelf": "center",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
        "width": "100%",
      }
    }
    testID="gift-card-list-category-scroll"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessibilityLabel="gift-card-list-category-searched-text"
          accessible={true}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#262C2F",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 16,
                "marginBottom": 20,
              },
            ]
          }
          testID="gift-card-list-category-searched-text"
        >
          
           Gift Cards
        </Text>
      </View>
      <View
        accessibilityLabel="gift-card-list-category-empty-state"
        accessible={true}
        style={
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-around",
            "paddingBottom": 28,
            "paddingLeft": 28,
            "paddingRight": 28,
            "paddingTop": 28,
          }
        }
        testID="gift-card-list-category-empty-state"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifySelf": "center",
              "paddingBottom": 28,
              "paddingLeft": 28,
              "paddingRight": 28,
              "paddingTop": 28,
            }
          }
        >
          BrandNoResults
          <Text
            style={
              Array [
                Object {
                  "color": "#262C2F",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 20,
                  "lineHeight": 30,
                },
                Object {
                  "marginTop": 20,
                  "textAlign": "center",
                },
              ]
            }
            type="titleHeader"
          >
            Woops! No search results :(
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#909BA0",
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Object {
                  "color": "#909BA0",
                  "fontSize": 18,
                  "marginTop": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Try with other words, or explore other brands in the reward section.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "marginBottom": 40,
          }
        }
      >
        <View
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
